### DISCUSSION TITLE
Figma Plugin

### DISCUSSION THREAD

**Discussion by X901** [bring-shrubbery/SVG-to-SwiftUI#34] [Ideas]
This will be awesome if it turn to **Figma Plugin**

so user can select svg icon , then tap on button to save it as `SwiftUI Shape Code` !
the Struct name will be the icon name or Frame name

**Comment by bring-shrubbery** [comment]
Thanks for the suggestion! I definitely should create a figma plugin for this - shouldn't be too hard 😅

**Comment by X901** [comment]
I did tried but I got this error

> (base) X901-Mac my-plugin % npm run build                             
> 
> > build
> > build-figma-plugin --typecheck --minify
> 
> 
> info Building plugin...
> error Module not found: Error: Package path ./lib/events is not exported from package /Users/X901/my-plugin/node_modules/@create-figma-plugin/utilities (see exports field in /Users/X901/my-plugin/node_modules/@create-figma-plugin/utilities/package.json)
> 


This is my Code

```
import { convert } from 'svg-to-swiftui-core';

export default function () {
    const nodes = figma.currentPage.selection;

    if (nodes.length === 0) {
        figma.notify('Please select an SVG layer.');
        figma.closePlugin();
        return;
    }

    const node = nodes[0];
    if (node.type !== 'VECTOR') {
        figma.notify('Selected layer is not an SVG.');
        figma.closePlugin();
        return;
    }

    node.exportAsync({ format: 'SVG' }).then(svgCodeArray => {
        const svgString = new TextDecoder().decode(svgCodeArray);
        const swiftUICode = convert(svgString);

        const structName = node.name;
        const finalCode = `struct ${structName}: View {
            ${swiftUICode}
        }`;

        figma.notify(`SwiftUI Code: ${finalCode}`);
        figma.closePlugin();
    });
}
```

**Comment by bring-shrubbery** [comment]
Hey @X901 

I looked into it and actually implemented the whole thing. I haven't used `build-figma-plugin` CLI that you used, but instead, I just used `tsup` and directly imported the code from the peer directory in this monorepo. I also made it into a Figma Codegen Plugin, so you can select the Vector element you want to convert, enter DEV mode, select SwiftUI export, and get the converted code.

You can check out how exactly I implemented it if you want. Let me know what you think, maybe I missed something?

**Reply by X901** [reply]
Awesome work!
I tried it, it working!

but I notice issue
when I add the code in SwiftUI project, it appeared upside down
but from website version it was appeared correctly

**Reply by bring-shrubbery** [reply]
hmmm, that's weird, might be `viewBox` issue, I'll check it later
